linkerd/app/core/src/errors.rs

```diff
@@ -26,15 +26,16 @@ impl<A, B: Default> respond::NewRespond<http::Request<A>> for NewRespond<B> {
     type Respond = Respond<B>;
 
     fn new_respond(&self, req: &http::Request<A>) -> Self::Respond {
-        if req.version() == http::Version::HTTP_2 {
-            let is_grpc = req
-                .headers()
-                .get(http::header::CONTENT_TYPE)
-                .and_then(|v| v.to_str().ok().map(|s| s.starts_with("application/grpc")))
-                .unwrap_or(false);
-            Respond::Http2 { is_grpc }
-        } else {
-            Respond::Http1(self.0)
+        match req.version() {
+            http::Version::HTTP_2 => {
+                let is_grpc = req
+                    .headers()
+                    .get(http::header::CONTENT_TYPE)
+                    .and_then(|v| v.to_str().ok().map(|s| s.starts_with("application/grpc")))
+                    .unwrap_or(false);
```

Comment:
Nit/TIOLI: might be slightly neater to express this as
```rust
let is_grpc = req
    .headers()
    .get(http::header::CONTENT_TYPE)
    .and_then(|v| v.to_str().ok())
    .iter()
    .contains(|s| s.starts_with("application/grpc"));
```
it's the same amount of code, but i think separating the conversion to a string and the testing the string for the predicate shows the flow more clearly.

admittedly, this doesn't really matter at all! just a thought